app/controllers/carto/api/visualizations_controller.rb

```diff
@@ -355,12 +355,16 @@ def count
         type = params.fetch(:type, 'datasets')
         typeList = (type == 'datasets') ? "'table','remote'" : "'derived'"
         categoryType = (type == 'datasets') ? 1 : 2
+        samples_user_id = sample_maps_user.id if categoryType == 2 && sample_maps_user
```

Comment:
Not strictly requesting a change, but adding a comment for readability:

It's not immediate to the reader that having a `samples_user_id` is tied to not only having `sample_maps_user` (which is the straightforward part) but to also `categoryType == 2` (not related to `sample_user` in any nature). Same happens above with the `samples` condition. 

With a software project this big and so many contributors, it's important to follow the "least surprise" principle, even at the expense of some extra lines :)